addons/mail/models/ir_attachment.py

```diff
@@ -13,9 +13,22 @@ def _post_add_create(self):
         """
         super(IrAttachment, self)._post_add_create()
         for record in self:
-            record.register_as_main_attachment()
+            record.register_as_main_attachment(force=False)
 
-    def register_as_main_attachment(self):
+    @api.multi
+    def unlink(self):
+        self.remove_as_main_attachment()
```

Comment:
I'm not sure we need to extract it in a method. 

We may also want to replace that by a ondelete='set null' in master